src/WebJobs.Extensions.DurableTask/LocalHttpListener.cs

```diff
@@ -59,7 +67,9 @@ public async Task StartAsync()
             int numAttempts = 1;
             do
             {
-                int availablePort = this.GetAvailablePort();
+                int availablePort = numAttempts == 1
+                    ? DefaultPort
+                    : this.portGenerator.Next(MinPort, MaxPort);
```

Comment:
Kind of nitpicky of me, but this might select the same port multiple times.  A better way would be to generate a random permutation and then iterate through it; then you'd be trying each port exactly once.